lang/java/reef-tang/tang/src/main/java/org/apache/reef/tang/implementation/java/ClassHierarchyImpl.java

```diff
@@ -204,15 +204,15 @@ public ClassHierarchyImpl(URL[] jars, Class<? extends ExternalConstructor<?>>[]
     if (root == null) {
       throw new NullPointerException();
     }
-    Node parent = root;
+    final Node parent = root;
 
-    Type argType = ReflectionUtilities.getNamedParameterTargetOrNull(clazz);
+    final Type argType = ReflectionUtilities.getNamedParameterTargetOrNull(clazz);
 
     if (argType == null) {
       return JavaNodeFactory.createClassNode(parent, clazz);
     } else {
 
-      @SuppressWarnings("unchecked")
```

Comment:
Place `final` on the same line as `NamedParameterNode<T>`? Moving the comment above the annotation might work.